ateam_kenobi/src/spatial/layers/their_defense_area_keepout.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright 2025 A Team
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#ifndef SPATIAL__LAYERS__THEIR_DEFENSE_AREA_KEEPOUT_HPP_
+#define SPATIAL__LAYERS__THEIR_DEFENSE_AREA_KEEPOUT_HPP_
+
+#include "spatial/spatial_layer_factory.hpp"
+
+namespace ateam_kenobi::spatial::layers
+{
+
+class TheirDefenseAreaKeepout : public SpatialLayerFactory {
+public:
+  TheirDefenseAreaKeepout()
+  : SpatialLayerFactory("TheirDefenseAreaKeepout") {}
+
+  void FillLayer(cv::Mat & layer, const World & world) override
+  {
+    SetupLayer(layer, CV_8UC1);
+
+    if(world.field.field_length == prev_field.field_length ||
```

Comment:
Shouldn't this be checking if all of the field dimensions are still the same (as in using and instead of or)?

Reply:
ssshhhhhhh.....
but also, yeah. Good thing to keep in mind for the real version of this system :)